### PR TITLE
[18339] Fix XML style

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
This PR fixes the following style issues:

* Extra blank lines at the end of the XML file
* Incorrect indentation for closing tags

This PR is built over #71 and must be merge after that one.

### CODE REVIEW

File: lib/src/cpp/utils/XMLManager.cpp [hunk 1]
```diff
@@ -236,6 +236,14 @@ void XMLManager::create_node(
 
     // Append new node to parent node
     parent_node->appendChild(last_node);
+
+    // Add text node to correctly set closing tag indentation
+    std::string dummy("\n");
```
Comment (JesusPoderoso) [inline lib/src/cpp/utils/XMLManager.cpp:241]: Perhaps this variable could be renamed into something like `fix_node_identation`

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
Tomorrow I'll test this but looks nice!
Does it fix also the EOF line breaks?

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]

File: lib/src/cpp/utils/XMLManager.cpp [hunk 2]
```diff
@@ -213,6 +213,7 @@ bool XMLManager::save_xml()
         // Config would configure serialized XML data
         config = serializer->getDomConfig();
         config->setParameter(xercesc::XMLUni::fgDOMWRTFormatPrettyPrint, true);
+        config->setParameter(xercesc::XMLUni::fgDOMWRTXercesPrettyPrint, false);
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply lib/src/cpp/utils/XMLManager.cpp:216]: No, it is not well [documented this setting](https://xerces.apache.org/xerces-c/apiDocs-3/classXMLUni.html#af5cb7cb41fa5648476f5aab2442adae4), but it seems that if the Pretty Print format is enabled, this setting is by default enabled too. Disabling this setting will remove the blank lines at the end of the file as explained in [this thread](https://lists.apache.org/thread/9dg14pdrh7dls5p7jw64tqcw7d7mq1yg).

File: lib/src/cpp/utils/XMLManager.cpp (hunk 1 above)
Comment (JLBuenoLopez-eProsima) [reply lib/src/cpp/utils/XMLManager.cpp]: In fact the string was already defined in `XMLManager` class: `LINE_BREAK`.

**Review by JesusPoderoso — DISMISSED** [review]
LGTM

File: lib/src/cpp/utils/XMLManager.cpp [hunk 3]
```diff
@@ -298,15 +298,23 @@ void XMLManager::create_node(
     reference_node =
             static_cast<xercesc::DOMNode*>(doc->createElement(xercesc::XMLString::transcode(tag_name.c_str())));
 
+    // Add text node to correctly set closing tag indentation
+    std::string node_formatting(LINE_BREAK);
+    for (int i = 0; i < depth_level - 1; i++)
+    {
+        node_formatting += "  ";
```
Comment (JesusPoderoso) [inline lib/src/cpp/utils/XMLManager.cpp:305]: Maybe we can include also a `constexpr static const char* NODE_IDENTATION = "  ";` in the Manager header too.
```suggestion
        node_formatting += NODE_IDENTATION;
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
I think you removed the commit that set the serializer configuration pretty format to `false` in the rebase process.

Plese, include it again

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Pull request merged**